MapboxNavigation/NavigationMapView.swift

```diff
@@ -533,19 +533,24 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         }
         
         if let lastLeg =  route.legs.last {
-            removeAnnotations(annotations ?? [])
-            let destination = MGLPointAnnotation()
+            removeAnnotations(annotationsToRemove() ?? [])
+            let destination = NavigationAnnotation()
             destination.coordinate = lastLeg.destination.coordinate
             addAnnotation(destination)
         }
     }
+    
+    func annotationsToRemove() -> [MGLAnnotation]? {
+        return annotations?.filter { type(of: $0) == NavigationAnnotation.self }
```

Comment:
Is there a need for NavigationAnnotation to be public? If it’s private, then we can filter on `$0 is NavigationAnnotation.self`.